### PR TITLE
Improvement/issue 10 add jest tests

### PR DESCRIPTION
**Pull request by Peter-holyoke**
## Related Issue

https://github.com/ProvidenceGeeks/website-frontend/issues/10 (and remaining items from #5)

## Summary of Changes

- [x] Add Jest
- [x] Write tests for bootstrap component
- [x] Write tests for home component
- [x] Write tests for hello world component
- [x] Add a test npm script

## Known Issues
https://github.com/ProvidenceGeeks/website-frontend/issues/12

### CODE REVIEW

**Comment by saimatsumoto** [comment]
resolves #10?

File: package.json [hunk 1]
```diff
@@ -23,24 +24,34 @@
...
+    "react-test-renderer": "^16.0.0",
     "sass-loader": "^6.0.6",
     "style-loader": "^0.19.0",
     "url-loader": "^0.6.2",
     "webpack": "^3.6.0",
     "webpack-dev-server": "^2.9.1",
     "webpack-md5-hash": "^0.0.5",
     "webpack-merge": "^4.1.0"
+  },
+  "jest": {
```
Comment (thescientist13) [inline package.json]: This seems to duplicate what's in _jest.config.js_ already, I think

File: src/react-raf-shim.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const raf = global.requestAnimationFrame = (cb) => {
```
Comment (thescientist13) [inline src/react-raf-shim.js]: if this is only for tests, maybe it makes sense to create a _shim_ folder in the test directory?  Also, maybe a comment to a link or github issue providing an explanation for why this is needed?

File: webpack.config.common.js [hunk 3]
```diff
@@ -8,6 +8,10 @@ module.exports = {
     vendor: './vendor.js'
   },
 
+  externals: {
```
Comment (thescientist13) [inline webpack.config.common.js]: could you provide some info for what this does?

File: package.json [hunk 4]
```diff
@@ -7,7 +7,8 @@
     "build": "npm run lint && webpack --config ./webpack.config.prod.js -p",
     "develop": "webpack-dev-server --open --config ./webpack.config.develop.js",
     "lint": "eslint *.js",
-    "serve": "npm run build && http-server ./build -o -p 9000"
+    "serve": "npm run build && http-server ./build -o -p 9000",
+    "test": "jest --coverage"
```
Comment (thescientist13) [inline package.json]: I wonder if it makes sense to only generate coverage when needed, as opposed to every time.  maybe a matter of perceived performance.  let's chat about approaches here.

Comment (thescientist13) [inline package.json]: We should add a test step here to [_config.yml_](https://github.com/ProvidenceGeeks/website-frontend/blob/master/.circleci/config.yml#L30) to make sure tests get run on every build

**Pull request merged**